File: examples/python/core/invoke/handle-timeout-error.py
from aws_durable_execution_sdk_python.config import Duration, InvokeConfig

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    """Handle timeout errors."""
    config = InvokeConfig(timeout=Duration.from_seconds(30))
    
    try:
        result = context.invoke(
            function_name="slow-function",
            payload=event,
            config=config,
        )
        return {"status": "success", "result": result}
    
    except CallableRuntimeError as e:
        if "timed out" in str(e).lower():
            context.logger.warning("Function timed out, using fallback")
            return {"status": "timeout", "fallback": True}
        raise

File: examples/java/core/invoke/handle-timeout-error.java
public class HandleTimeoutErrorHandler extends DurableHandler<Map<String, Object>, Map<String, Object>> {

    @Override
    public Map<String, Object> handleRequest(Map<String, Object> event, DurableContext ctx) {
        // Handle timeout errors
        InvokeConfig config = InvokeConfig.builder()
            .timeout(Duration.ofSeconds(30))
            .build();

        try {
            var result = ctx.invoke(
                "slow_operation",
                "slow-function",
                event,
                Map.class,
                config
            );
            return Map.of("status", "success", "result", result);

        } catch (InvokeTimedOutException e) {
            ctx.getLogger().warn("Function timed out, using fallback");
            return Map.of("status", "timeout", "fallback", true);

        } catch (InvokeFailedException e) {
            throw e;
        }
    }
}
